### ISSUE TITLE
Rule suggestion: `vue/prefer-single-payload-in-event`

### ISSUE THREAD

**Issue by FloEdelmann** [vuejs/eslint-plugin-vue#2005]
**Please describe what the rule should do:**
When emitting an event, one can pass an event payload. This can consist of multiple parameters. However, when using the `always` style in the event handler, one can only receive the first event payload parameter.

```vue
<my-button @my-event="myHandler($event)" />
```

See also the [`vue/v-on-function-call` rule](https://eslint.vuejs.org/rules/v-on-function-call.html) and https://github.com/vuejs/eslint-plugin-vue/issues/2001#issuecomment-1269475268.

**What category should the rule belong to?**
[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```ts
// BAD:
this.$emit('my-event', foo, bar)
emit('my-event', foo, bar)
defineEmits({ 'my-event': (foo, bar) => true })
defineEmits({ 'my-event': (foo: string, bar: string) => true })
defineEmits<{ (e: 'my-event', foo: string, bar: string): void }>()
{ emits: { 'my-event': (foo, bar) => true } }
{ emits: { 'my-event': (foo: string, bar: string) => true } }

// GOOD:
this.$emit('my-event', foo)
emit('my-event', foo)
defineEmits(['my-event'])
defineEmits({ 'my-event': null })
defineEmits({ 'my-event': (foo) => true })
defineEmits({ 'my-event': (foo: string) => true })
defineEmits<{ (e: 'my-event', foo: string): void }>()
{ emits: ['my-event'] }
{ emits: { 'my-event': null }
{ emits: { 'my-event': (foo) => true } }
{ emits: { 'my-event': (foo: string) => true } }
```
